docs/conf.py

```diff
@@ -439,6 +426,152 @@ def resolve_astropy_and_dev_reference(app, env, node, contnode):
+    # Coordinates
+    "EarthLocation": ":class:`~astropy.coordinates.EarthLocation`",
+    "Angle": "`~astropy.coordinates.Angle`",
+    "Latitude": "`~astropy.coordinates.Latitude`",
+    "Longitude": ":class:`~astropy.coordinates.Longitude`",
+    "BaseFrame": "`~astropy.coordinates.BaseCoordinateFrame`",
+    "SkyCoord": ":class:`~astropy.coordinates.SkyCoord`",
+    "SpectralCoord": "`~astropy.coordinates.SpectralCoord`",
+    # Cosmology
+    "Cosmology": ":class:`~astropy.cosmology.Cosmology`",
+    "Cosmology.read": ":meth:`~astropy.cosmology.Cosmology.read`",
+    "Cosmology.write": ":meth:`~astropy.cosmology.Cosmology.write`",
+    "Cosmology.from_format": ":meth:`~astropy.cosmology.Cosmology.from_format`",
+    "Cosmology.to_format": ":meth:`~astropy.cosmology.Cosmology.to_format`",
+    "FLRW": ":class:`~astropy.cosmology.FLRW`",
+    "LambdaCDM": ":class:`~astropy.cosmology.LambdaCDM`",
+    "FlatLambdaCDM": ":class:`~astropy.cosmology.FlatLambdaCDM`",
+    "WMAP1": ":ref:`astropy_cosmology_realizations_WMAP1`",
+    "WMAP3": ":ref:`astropy_cosmology_realizations_WMAP3`",
+    "WMAP5": ":ref:`astropy_cosmology_realizations_WMAP5`",
+    "WMAP7": ":ref:`astropy_cosmology_realizations_WMAP7`",
+    "WMAP9": ":ref:`astropy_cosmology_realizations_WMAP9`",
+    "Planck13": ":ref:`astropy_cosmology_realizations_Planck13`",
+    "Planck15": ":ref:`astropy_cosmology_realizations_Planck15`",
+    "Planck18": ":ref:`astropy_cosmology_realizations_Planck18`",
+    "FlatCosmologyMixin": ":class:`~astropy.cosmology.FlatCosmologyMixin`",
+    "FlatFLRWMixin": ":class:`~astropy.cosmology.FlatFLRWMixin`",
+    "default_cosmology": ":class:`~astropy.cosmology.default_cosmology`",
+    # SAMP
+    "SAMPClient": ":class:`~astropy.samp.SAMPClient`",
+    "SAMPIntegratedClient": ":class:`~astropy.samp.SAMPIntegratedClient`",
+    "SAMPHubServer": ":class:`~astropy.samp.SAMPHubServer`",
+    "SAMPHubProxy": ":class:`~astropy.samp.SAMPHubProxy`",
+    # Table
+    "Column": ":class:`~astropy.table.Column`",
+    "MaskedColumn": ":class:`~astropy.table.MaskedColumn`",
+    "TableColumns": ":class:`~astropy.table.TableColumns`",
+    "Row": ":class:`~astropy.table.Row`",
+    "Table": ":class:`~astropy.table.Table`",
+    "QTable": ":class:`~astropy.table.QTable`",
+    # Time
+    "Time": ":class:`~astropy.time.Time`",
+    "TimeDelta": ":class:`~astropy.time.TimeDelta`",
+    # Timeseries
+    "TimeSeries": ":class:`~astropy.timeseries.TimeSeries`",
+    "BinnedTimeSeries": ":class:`~astropy.timeseries.BinnedTimeSeries`",
+    # Distribution
+    "Distribution": ":class:`~astropy.uncertainty.Distribution`",
+    # Units
+    "PhysicalType": ":class:`~astropy.units.PhysicalType`",
+    "Quantity": ":class:`~astropy.units.Quantity`",
+    "Unit": ":class:`~astropy.units.UnitBase`",
+    "StructuredUnit": ":class:`~astropy.units.StructuredUnit`",
+    # Utils
+    "Masked": ":class:`~astropy.utils.masked.Masked`",
+    # Minimum versions
```

Comment:
I think we lost the ability to be able to dynamically insert new minversion listing for dependencies (which tend to keep growing nowdays).

That said, `docs/conf.py` is easier to discover than `docs/common_links.txt` in general, plus the speed up. So the benefit of this PR outweighs the risk of forgetting to insert a new entry here.